File: 2050-count-good-numbers/count-good-numbers.java
class Solution {
    public int countGoodNumbers(long n) {
        long even = (n+1)/2;
        long odd = n/2;
        long MOD = 1000000007;
        long evenPart = pow(5,even);
        long oddPart = pow(4,odd);
        

        return (int) ((evenPart * oddPart ) % MOD);
        
    

    }
    public long pow(long a , long b){ 
        long MOD = 1000000007;
        if(b ==0) return 1;
        long half = pow(a , b/2);
        if( b %2 == 0) return (half * half )% MOD;
        return (half * half * a )% MOD;

    }
}

File: 2050-count-good-numbers/count-good-numbers.py
class Solution:
    def countGoodNumbers(self, n: int) -> int:
        
        mod = 10** 9 + 7 

        oddpos = (n) // 2
        evenpos = (n + 1 ) // 2

        odd = pow(4  , oddpos , mod) 
        even = pow(5 , evenpos , mod)

        return (even * odd) % mod